### ISSUE TITLE
Integrating Piecewise functions with simple trig

### ISSUE THREAD

**Issue by liamwheen** [sympy/sympy#20449]
The issue arises when integrating
`Max(0, sin(1)*sin(gamma) - 0.5*cos(1)*cos(gamma))`, 
The integration seems to take a very long time, however, using `evalf()` beforehand makes it much faster.
This gives `Max(0, 0.841470984807897*sin(gamma) - 0.27015115293407*cos(gamma))`.

Integrating this should give 1.767546394, but it actually gives 0. Only by changing the coefficients slightly allows it to work out the correct answer. It seems like some issue with the way sin(1) and cos(1) are being evaluated.

**Comment by oscarbenjamin** [comment]
The problem is that solveset can be slow for some cases like this. Something like this is being called (which is very slow):
```python
solveset(sin(x) + sin(1)*cos(x), x, Interval(0, 2*pi))
```

**Comment by liamwheen** [comment]
Does that explain why it evaluates the integral to be 0?

**Comment by oscarbenjamin** [comment]
No, I haven't investigated where the zero comes from. My guess would be that something went wrong trying to solve for the limits and it ended up just integrating the 0.
